### PR TITLE
update contract workflows to take in external artifacts

### PR DESCRIPTION
**Pull request by RnkSngh**
@zhengyangfeng00 Added support for allowing deploying external contracts in our in-js code, but this PR adds support for integrating this functionality through our bin scripts run using `npx`

## Release Notes

- **New Features**
  - Introduced support for additional contract factories and artifacts in transaction processing.
  - Added constants for extra bindings and artifacts paths to enhance contract management.
  - Enhanced multisig account management with new schemas and validation functions.
  - Improved flexibility in transaction processing and contract updates with new parameters and object structures.
  - Updated contract verification to utilize structured input types for enhanced detail.
  - Simplified the deployment process with improved account registry handling.
  - Enhanced the `readArtifactFile` function to accept a Solidity compiler version for better artifact management.

- **Bug Fixes**
  - Improved error handling and messaging consistency across various functions.
  - Added validation checks for multisig account initialization.
  - Enhanced error handling for unknown account types during transaction processing.
  - Improved error handling for contract verification based on the contract's specific compiler version.

- **Documentation**
  - Updated comments for clarity regarding the new parameters and functionality.

- **Style**
  - Standardized string formatting to single quotes for consistency throughout the codebase.

- **Chores**
  - Updated package version from `4.0.5` to `4.0.7`.

### CODE REVIEW

File: src/scripts/update-contracts-script.ts [hunk 1]
```diff
@@ -8,20 +8,34 @@
...
     accounts.mustGet(chain.chainName),
     ContractRegistryLoader.emptySingle(),
     contractUpdates,
     getOutputLogger(),
-    false
+    false,
+    false,
+    true,
+    extraContractFactories,
+    extraArtifactsPath,
```
[the remark this answers is not in the document]
Comment (dshiell) [reply src/scripts/update-contracts-script.ts:38]: +1

**Review by dshiell — COMMENTED** [review]
maybe review and resolve the ai suggestions and look into the ci failures?

**Comment by RnkSngh** [comment]
Ci failures are due to stacking PRs - should be passing once we have all into one

File: src/scripts/update-contracts-script.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (RnkSngh) [reply src/scripts/update-contracts-script.ts]: good one!

File: src/utils/io.ts [hunk 2]
```diff
@@ -173,34 +175,43 @@ export function setStdoutStderr(
...
+    path.join(basePath, `${artifactName}.sol`, `${artifactName}.json`)
+  );
+  for (const path of paths) {
+    try {
+      return await fsAsync.readFile(path, 'utf8');
+    } catch (e) {
+      console.error(`error reading from file in extra file ${path}: \n`, e);
+    }
   }
+  return '';
```
[the remark this answers is not in the document]
Comment (RnkSngh) [reply src/utils/io.ts]: already fixed in a subsequent commit

**Review by dshiell — APPROVED** [review]
lets goooooo!

**Pull request merged**